backend/tests/prompts/prompting_test.py

```diff
@@ -63,18 +58,14 @@ def test_load_best_next_step_template():
 AGENT LIST:
 
 
-If none of the agents are appropriate for solving this goal, choose "UnresolvableTaskAgent".
-If you believe you have solved the final problem, choose "GoalAchievedAgent"
-
-
 ## Determine the next best step
 Your task is to pick one of the mentioned agents above to complete the task.
-If you\'re last attempt at the Current Task looks successful, move on to the Next Task
+If from the history you see that the agent has failed to complete a task (by repeating a task with the same agent), do not assign the task to them again.
```

Comment:
This part of the prompt may need expanding or highlighting further. Or even perhaps we should make choosing the same agent a hard-coded blocker (remove it from the list of agents to choose from the second time for example).

My first run of the example prompt had the Netflix transactions call MathsAgent 5 times
```
INFO:     Final scratchpad:
INFO:     [
    {
        "agent_name": "DatastoreAgent",
        "task": "Find all transactions on Netflix from the user's account last month.",
        "result": "Last month you spend \u00a36.99 on Netflix"
    },
    {
        "agent_name": "DatastoreAgent",
        "task": "Find all transactions on Amazon from the user's account last month.",
        "result": "Last month you spend \u00a364.21 on Amazon"
    },
    {
        "agent_name": "MathsAgent",
        "task": "Sum up the transactions for Netflix.",
        "result": "You have spent more on Netflix transaction 1 (15.99) than Netflix transaction 2 (13.99) in the last month"
    },
    {
        "agent_name": "MathsAgent",
        "task": "Sum up the transactions for Netflix.",
        "result": "You have spent more on Netflix transaction 1 (15.99) than Netflix transaction 2 (13.99) in the last month"
    },
    {
        "agent_name": "MathsAgent",
        "task": "Sum up the transactions for Netflix.",
        "result": "You have spent more on Netflix transaction 1 (15.99) than Netflix transaction 2 (13.99) in the last month"
    },
    {
        "agent_name": "MathsAgent",
        "task": "Sum up the transactions for Netflix.",
        "result": "You have spent more on Netflix transaction 1 (15.99) than Netflix transaction 2 (13.99) in the last month"
    },
    {
        "agent_name": "MathsAgent",
        "task": "Sum up the transactions for Netflix.",
        "result": "You have spent more on Netflix transaction 1 (15.99) than Netflix transaction 2 (13.99) in the last month"
    },
    {
        "agent_name": null,
        "task": "Sum up the transactions for Netflix.",
        "result": "I am sorry, but I was unable to find an answer to this task"
    },
    {
        "agent_name": null,
        "task": "Sum up the transactions for Amazon.",
        "result": "I am sorry, but I was unable to find an agent to solve this task"
    },
    {
        "agent_name": "MathsAgent",
        "task": "Compare the total spending on Netflix and Amazon.",
        "result": "You have spent more on Netflix (1500) than Amazon (1200) in the last month"
    }
]
```

If this is an anomaly or not I'm not sure, but it seems it will pick the same agent multiple times without any other good choices

Reply:
I'll try a tweaked version